### PR TITLE
DCS-818 Get list of courts - add Jurisdiction Codes filter.

### PR DESCRIPTION
**Pull request by bell-pa**

### CODE REVIEW

File: src/main/resources/db/migration/data/R__2_1__AGENCY_LOCATIONS.sql [hunk 1]
```diff
@@ -14,10 +14,13 @@ INSERT INTO AGENCY_LOCATIONS (AGY_LOC_ID, DESCRIPTION, AGENCY_LOCATION_TYPE, ACT
...
-  VALUES ('COURT1', 'Court 1', 'CRT', 'Y');
-INSERT INTO AGENCY_LOCATIONS (AGY_LOC_ID, DESCRIPTION, AGENCY_LOCATION_TYPE, ACTIVE_FLAG)
-  VALUES ('ABDRCT', 'Court 2', 'CRT', 'Y');
+
+INSERT INTO AGENCY_LOCATIONS (AGY_LOC_ID, DESCRIPTION, AGENCY_LOCATION_TYPE, ACTIVE_FLAG, JURISDICTION_CODE)
+  VALUES ('COURT1', 'Court 1', 'CRT', 'Y', 'MC');
+
+INSERT INTO AGENCY_LOCATIONS (AGY_LOC_ID, DESCRIPTION, AGENCY_LOCATION_TYPE, ACTIVE_FLAG, JURISDICTION_CODE)
+  VALUES ('ABDRCT', 'Court 2', 'CRT', 'Y', 'CC');
+
```
Comment (GurnankCheema) [inline src/main/resources/db/migration/data/R__2_1__AGENCY_LOCATIONS.sql:23]: Do we need a statement for immigration courts?

**Review by GurnankCheema — DISMISSED** [review]
LGTM

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Comment by bell-pa** [comment]
Jurisdiction codes to select the required courts:
* CC - crown courts
* MC - Magistrates' courts
* IMM - Immigration courts

To test this in 'dev':
{code:bash}
WHEREABOUTS_SECRET=$(kubectl -n whereabouts-api-dev get secret whereabouts-api -o json | jq ".data.OAUTH_CLIENT_SECRET | @base64d")

WHEREABOUTS_KEY=$(echo -n "whereabouts-api-client:$WHEREABOUTS_SECRET"| base64)

WHEREABOUTS_TOKEN=$(curl -X POST "https://sign-in-dev.hmpps.service.justice.gov.uk/auth/oauth/token?grant_type=client_credentials" -H 'Content-Type: application/json' -H 'Content-Length: 0' -H "Authorization: Basic $WHEREABOUTS_KEY" | jq -r '.access_token')

curl -X GET 'https://api-dev.prison.service.justice.gov.uk/api/agencies/type/CRT?jurisdictionCode=IMM&jurisdictionCode=CC&jurisdictionCode=MC' -H "Authorization: Bearer $WHEREABOUTS_TOKEN"
{code}

As usual you will need curl and kubernetes set up for this to work.
Just pipe into jq to count the number of courts:
{code:bash}
 | jq 'length'
{code}

**Pull request merged**